### PR TITLE
Add data store and store window bounds

### LINKED ISSUE #24: Window size is not saved
Steps to reproduce:
- Open Signal Muon
- Resize the window to your needs
- Close Signal Muon
- Open it again

Expected Result: The window has the size that was previously set by the user
Actual Result: Window is again 800x700

@diracdeltas are those issues welcomed or not needed/wished?

### PR DESCRIPTION
**Pull request by mammuth**
I added a generic store handler, which still stores production and dev settings in different files. It also can handle defaults.

Right now it's only used for storing the window bounds. Bounds are updated when the window is closed.

Are you fine with handling userData this way, @diracdeltas ?
There are no linting errors.

Fixes diracdeltas/signal-muon/issues/24

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -16,10 +17,13 @@ const signalExtensionPath = process.env.NODE_ENV === 'production'
   ? path.join(__dirname, '..', '..', 'Signal-Desktop')
   : path.join(__dirname, 'Signal-Desktop')
 
-// use a separate data directory for development
```
Comment (diracdeltas) [inline main.js]: This should be undeleted I think

Comment (diracdeltas) [reply main.js]: nevermind, i see where it is handled below

File: store.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+const electron = require('electron')
+const path = require('path')
+const fs = require('fs')
+
+class Store {
+  constructor (opts) {
+    const isProduction = process.env.NODE_ENV !== 'development'
+    // Use a different userData file for development
```
Comment (diracdeltas) [inline store.js:8]: i think you still need to do app.setPath here; otherwise prod and dev will share the same app data directory.

File: store.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    console.log('Using userData located at ' + this.path)
+  }
+
+  get (key) {
+    return this.data[key]
+  }
+
+  set (key, val) {
+    this.data[key] = val
+    fs.writeFileSync(this.path, JSON.stringify(this.data))
```
Comment (diracdeltas) [inline store.js]: no need to use writeFileSync instead of writeFile here

**Comment by diracdeltas** [comment]
thanks for this! this seems like a reasonable approach, assuming that https://github.com/diracdeltas/signal-desktop isn't saving the window sizes. (If so, I would prefer to get them from the signal-desktop extension in order to avoid having an app data store at all.)

**Comment by mammuth** [comment]
Thanks for the fast review. I changed the PR based on your comments.

File: store.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+const electron = require('electron')
+const path = require('path')
+const fs = require('fs')
+
+class Store {
+  constructor (opts) {
+    const isProduction = process.env.NODE_ENV !== 'development'
+    if (!isProduction) {
+      app.setPath('userData', path.join(app.getPath('appData'), 'signal-muon'))
```
Comment (diracdeltas) [inline store.js]: app is not defined here. i think it is better to put this where it was before in main.js. The path does not need to be set every time that a store is instantiated.

File: store.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    }
+    // Use a different userData file for development
+    const configName = isProduction ? opts.configName : opts.configName + '-dev'
```
Comment (diracdeltas) [inline store.js]: it's unnecessary to use a different configName now that the appData directories are different

File: store.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    // Renderer process has to get `app` module via `remote`, whereas the main process can get it directly
+    const userDataPath = (electron.app || electron.remote.app).getPath('userData')
```
Comment (diracdeltas) [inline store.js]: probably safe to assume for now that this will only get called in the main process

**Comment by diracdeltas** [comment]
please `git rebase -i` to squash commits into one commit. thanks!

**Comment by mammuth** [comment]
Done. 
You're no big fan of using GitHub's shiny new button for Squashing and Merging directly from PR's?😉

**Comment by diracdeltas** [comment]
@mammuth actually i forgot it existed. thanks for the reminder!

**Review by diracdeltas — CHANGES_REQUESTED** [review]

**Comment by diracdeltas** [comment]
^ ping on requested changes

**Comment by mammuth** [comment]
Wups, didn't notice that you requested new changes 😊 sorry for the delay.

**Review by diracdeltas — APPROVED** [review]

**Pull request merged**